files/scripts/machine_provision.sh

```diff
@@ -0,0 +1,67 @@
+#!/bin/bash
+
+if [ "$#" -gt 16 ] || [ "$#" -lt 12 ]; then
+	echo "Illegal number of arguments !"
+	echo "machine_provision.sh [WORKSPACE] [machine_list] [machine_type] [job_type] [scripts_branch] [machine_arch] [machine_subarch] [kernel_opts] [kernel_desc] [initrd_desc] [preseed_name] [preseed_type] kernel_path initrd_path preseed_path -v"
+	echo "[required] optional"
+	exit 1
+fi
+
+if [ "${16}" == '-v' ]; then
```

Comment:
if you allow 12~16 arguments, this will be anything from 12th to 16th
shouldn't -v allowed in any position?
the best way here is to use getopts and force --option=value for everything but obvious things like workspace.